watson-developer-cloud/discovery-components
# fix: search facet categories bug

#### What do these changes do/fix?

https://github.ibm.com/Watson-Discovery/the-tooling-core-tracker/issues/1544

Fixes a bug found during integration testing where facets will not load in certain double nested situations when the field name is `enriched_text.entities.text`. This is because these dictionary facets were being read as having categories when they did not.

#### How do you test/verify these changes?
~~Run this against `weary-griffin` Sample Project right now and see that facets will load. See that they currently not without this.~~

If that is gone, the same situation is now reflected as a test, see that tests pass. Remove the change in `FieldFacets.tsx` and see that the test will fail to load the facet group.

#### Have you documented your changes (if necessary)?
Tests have been updated.

#### Are there any breaking changes included in this pull request?
No
